src/include/parser/expression/expression_util.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <algorithm>
+#include <cstdlib>
+#include <set>
+#include <string>
+
+#include "catalog/catalog_defs.h"
+#include "common/managed_pointer.h"
+#include "parser/expression/abstract_expression.h"
+#include "parser/expression/column_value_expression.h"
+
+namespace terrier::parser::expression {
+
+/**
+ * An utility class for Expression objects
+ */
+class ExpressionUtil {
+ public:
+  // Static utility class
+  ExpressionUtil() = delete;
+
+  /**
+   * Populate the given set with all of the column oids referenced
+   * in this expression tree.
+   * @param col_oids place to store column oids in the expression
+   * @param expr to extract column oids from
+   */
+  static void GetColumnOids(std::set<catalog::col_oid_t> *col_oids, common::ManagedPointer<AbstractExpression> expr) {
```

Comment:
What was the reason to use `std::set`? I actually think this should be a `std::vector`. The main reason is because a lot of our functions that use `col_oid_t` store them in an `std::vector` ([`SqlTable::InitializerForProjectedRow`](https://github.com/cmu-db/terrier/blob/94e999bfb9355dafb03f15b6c86803650599fe37/src/include/storage/sql_table.h#L180) most likely being the most popular). Converting from an `std::set` to `std::vector` is O(n) in c++ so we should avoid that if possible. 

If for some reason we do need a set, I can't think of a reason we need the ordering `std::set` provides, so we could just use `std::unordered_set`

Reply:
Is it possible for ```col_oid_t``` to be duplicated? I forked this from older code that was using a ```std::set```, but I am not entirely sure about the behavior. If duplicates are not a problem I can switch to ```std::vector```?

Reply:
I don't think this is supposed to be ordered.

@17zhangw Can you point us to the code that would use this information?

Reply:
Yes, a `col_id_t` can be duplicated in an expression tree:

```
SELECT * FROM xxx WHERE id = 1 OR id = 2
```

Reply:
Technically, it would be used within this file:
```
https://github.com/cmu-db/terrier/blob/optimizer/src/include/optimizer/index_util.h
```

The current optimizer (in the branch) only supports picking indexes built only on base columns. But I do anticipate we'll be needing this function in the near future.